lib/routemaster/resources/rest_resource.rb

```diff
@@ -18,8 +18,13 @@ def show(id=nil)
         @client.get(@url.gsub('{id}', id.to_s))
       end
 
-      def index
-        @client.get(@url)
+      def index(params: {}, filters: {})
+        params_and_filters = params.merge(filters)
+        if params_and_filters == {}
+          @client.get(@url)
+        else
+          @client.get(@url, params: filters.merge(params))
+        end
```

Comment:
would it be easier to just do this?
```ruby
def index(params: {}, filters: {})
  hateoas_response = @client.get(@url, params: params.merge(filters))

  EnumerableHateoasResponse.new(response: hateoas_response)
end
```

1) Passing an empty hash to the params key shouldn't really affect how the get request behaves (I might be mistaken)
2) Instead of defining a constructor in `hateoas_response.rb` and parsing the URL you could just move the responsibility here, in that way you'll never be mistaken

What do you think?